### PR TITLE
Add support for Catbox

### PR DESCRIPTION
**Pull request by henrik9999**
example https://catbox.moe/c/gr4t8i

### CODE REVIEW

**Comment by Algoinde** [comment]
I don't remember, was Catbox fine with using them for Cubari galleries?

**Comment by Algoinde** [comment]
@funkyhippo @Einlion could you take a look? Looks LGTM to me

File: proxy/sources/catbox.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    def catbox_common(self, meta_id: str) -> Optional[Dict]:
+        resp = post_wrapper(f"https://catbox.moe/user/api.php", data={"reqtype": "getalbum", "short": meta_id})
+
+        if resp.status_code != 200:
+            return None
+
+        json = resp.json()
+
+        if json["status"] != 200 or not json["success"]:
+            return None
```
Comment (funkyhippo) [inline proxy/sources/catbox.py:31]: Huh, this was an incredibly difficult API to track down. There's virtually no documentation, and the first reference I found was in the [node library](https://github.com/tentaclenotsoft/node-catbox/blob/ac1844f8307a6b645ce64dd2eb40592f3ef684cd/lib/structures/Catbox.js#L53-L68) linked on [their tools doc](https://catbox.moe/tools.php).

Anyway, this is a strange API:
* The returned content-type is `text/html`, despite the body being a JSON body
* The JSON includes a `status` and `success` field that you're checking -- it's unclear whether the HTTP status code is correctly used here
* Creating an album requires a dance of uploading the individual files, then creating a collection with the shortcodes after you've uploaded all your files
* There's no validation for the album creation -- you can pass in any arbitrary string of images

```bash
# Create an "album"
curl -F "reqtype=createalbum" -F 'title=title' -F 'desc=desc' -F 'files="a b c d e f"' https://catbox.moe/user/api.php

# From the shortcode above
curl -F "reqtype=getalbum" -F "short=0j1hb2" https://catbox.moe/user/api.php
```

Weird.

**Review by funkyhippo — APPROVED** [review]
I'm peaceful with this change, thanks for the contribution.

**Pull request merged**